jbmc/src/java_bytecode/java_types.cpp

```diff
@@ -120,15 +120,24 @@ reference_typet java_array_type(const char subtype)
   return java_reference_type(symbol_type);
 }
 
-/// Return the type of the elements of a given java array type
-/// \param array_type The java array type
-/// \return The type of the elements of the array
-typet java_array_element_type(const symbol_typet &array_type)
+/// Return a const reference to the element type of a given java array type
+/// \param array_symbol The java array type
+const typet &java_array_element_type(const symbol_typet &array_symbol)
 {
   INVARIANT(
-    is_java_array_tag(array_type.get_identifier()),
+    is_java_array_tag(array_symbol.get_identifier()),
     "Symbol should have array tag");
-  return array_type.find_type(ID_C_element_type);
+  return array_symbol.find_type(ID_C_element_type);
+}
+
+/// Return a non-const reference to the element type of a given java array type
+/// \param array_symbol The java array type
+typet &java_array_element_type(symbol_typet &array_symbol)
+{
+  INVARIANT(
+    is_java_array_tag(array_symbol.get_identifier()),
+    "Symbol should have array tag");
+  return const_cast<typet &>(array_symbol.find_type(ID_C_element_type));
```

Comment:
Using `const_cast` doesn't seem nice. I think you should use `add_type` instead (it's true the name can be misleading, like `add`, I think they should be renamed `get_type` and `get`, and `get` should be `find_id`)

Reply:
This should use `add_type`.